modules/lang-mustache/src/main/java/org/opensearch/script/mustache/SearchTemplateRequest.java

```diff
@@ -56,7 +58,7 @@
 /**
  * A request to execute a search based on a search template.
  */
-public class SearchTemplateRequest extends ActionRequest implements CompositeIndicesRequest, ToXContentObject {
+public class SearchTemplateRequest extends ActionRequest implements IndicesRequest.Replaceable, CompositeIndicesRequest, ToXContentObject {
```

Comment:
@cwperks why do you need replaceable here? From the description, the issue is that indices could not be extracted, in this case `implements IndicesRequest` is fully sufficient,

Reply:
Also tagging @peternied

Reply:
If I'm understanding the flow in IndexReplacerResolver, the replacements only come into the picture when dnfof is enabled.

dnfof === do not fail on forbidden. Looking into that more now to see if that's applicable for these requests too.

Reply:
Oh I get it now. With `dnfof` enabled the security plugin will go through all roles mapped to a user and compare the indices from the request against with the list of indices the user is permitted to for the given action. The security plugin modifies the request to only include the indices from the request that the user is permitted to for the given action.

For instance let's say I have user `craig` that is permitted to do `indices:data/read/search*` on `logs`, but not on `movies`. 

1. With dnfof disabled, the follow request would fail because `craig` does not have permission to search `movies` index even though it can search `logs` index:

```
curl -XGET https://craig:TestPassword123_@localhost:9200/logs,movies/_search/template --insecure -H "Content-Type: application/json" --data '
{
  "source": {
    "query": {
      "match": {
        "title": "{{title}}"
      }
    }
  },
  "params": {
    "title": "Titanic"
  }
}
'
```

2. With dnfof enabled the request would work, but it would only in effect to the query against `logs` index and the logic in the security plugin replaces the index before the search is performed.

Reply:
@reta ^

Reply:
@cwperks 

> @nibix I know you mentioned a bit about adding `IndicesRequest` into `IndexResolverReplacer.getOrReplaceAllIndices` before, but I see its still not in the big `if { } else { }` block. Do you know if there is a good reason why it is not handled already because `Replaceable` is.

Sorry, I am a bit late to the game. Generally, I do not think there is a good reason why IndicesRequest is not handled here. Most likely it was just not needed so far. 

Just as a BTW/FYI, another place in the security plugin which utilizes the Replaceable interface is the PrivilegesInterceptor:

https://github.com/opensearch-project/security/blob/0d915e24cf16913f01d7d97a9a007c99b39bc2ed/src/main/java/org/opensearch/security/configuration/PrivilegesInterceptorImpl.java#L349